native/acter/src/api/common.rs

```diff
@@ -544,3 +547,18 @@ impl ThumbnailSize {
         self.height
     }
 }
+
+pub fn new_thumb_size(width: u64, height: u64) -> Result<ThumbnailSize> {
+    ThumbnailSize::new(width, height)
+}
+
+pub(crate) fn into_media_format(thumb_size: Option<Box<ThumbnailSize>>) -> MediaFormat {
+    match thumb_size {
+        Some(thumb_size) => MediaFormat::Thumbnail(MediaThumbnailSize {
+            method: get_content_thumbnail::v3::Method::Scale,
+            width: thumb_size.width(),
+            height: thumb_size.height(),
+        }),
+        None => MediaFormat::File,
+    }
+}
```

Comment:
let's move this onto `ThumbnailSize` and split it up please:

```suggestion
impl ThumbnailSize {
   pub fn into_media_format(&self) -> MediaFormat {
       MediaFormat::Thumbnail(MediaThumbnailSize {
            method: get_content_thumbnail::v3::Method::Scale,
            width: self.width(),
            height: self.height(),
        }
   }
   
   pub fn parse_into_media_format(thumb_size: Option<Box<ThumbnailSize>>) -> MediaFormat {
        match thumb_size {
            Some(thumb_size) => thumb_size.into_media_format(),
            None => MediaFormat::File,
       }
    }
   
```

and then use `ThumbnailSize::parse_into_media_format(a)`  and `thumbnail_size.into_media_format()` for calling it

Reply:
Fixed by https://github.com/acterglobal/a3/pull/1246/commits/518193fa236e942607e738383ba8ffb64b96b9ca